modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -26,6 +29,20 @@
 
 (set! *warn-on-reflection* true)
 
+(defn- sql->lines [sql]
+  (str/split-lines (mdb.query/format-sql sql :snowflake)))
+
+(defn- query->native [query]
+  (let [native-query (atom nil)
+        check-sql-fn (fn [_ _ sql & _]
+                       (reset! native-query sql)
+                       (throw (Exception. "done")))]
+    (with-redefs [sql-jdbc.execute/prepared-statement check-sql-fn
+                  sql-jdbc.execute/execute-statement! check-sql-fn]
+      (u/ignore-exceptions
+        (qp/process-query query))
+      @native-query)))
```

Comment:
Might be simpler just to throw an exception with `sql` as a parameter (see `ex-info`) and catch it.